subsys/bluetooth/host/adv.c

```diff
@@ -520,14 +550,18 @@ static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op
 				}
 
 				if (offset == 0) {
-					set_data[data_offset + 0] = len + 1;
-					set_data[data_offset + 1] = type;
-					memcpy(&set_data[data_offset + 2], data[j].data,
-					       set_data[len_offset] - 2);
-					offset += set_data[len_offset] - 2;
+					size_t data_len_offset = data_offset;
+					size_t data_type_offset = data_offset + 1;
+					size_t data_value_offset = data_offset + 2;
+
+					set_data[data_len_offset] = len + 1;
+					set_data[data_type_offset] = type;
+					(void)memcpy(&set_data[data_value_offset], data[j].data,
+						     set_data[data_type_offset] - 2);
```

Comment:
Typo here? Should it be `len_offset` instead of `data_type_offset`?

Reply:
Yes, you are right